update README.md and the navigation-timing.js file

ryanpoplin opened the pull request:
# What is this and why is it required?

These updates were made to interface with Segment.com services

Commit:
update README.md and the navigation-timing.js file

This updates were made to interface with Segment.com via this file.

Reviewed-By: Ryan Poplin <ryan.poplin@turner.com>
Fix: feature/CNNWE-828

Commit:
add error handling for Segment globals

I've added a log message to denote a failure in accessing the required
global variables for segment interface.

Reviewed-By: Ryan Poplin <ryan.poplin@turner.com>
Fix: feature/CNNWE-828

adslaton commented on src/navigation-timing.js:
```diff
@@ -97,14 +97,28 @@ window.NTAPI = window.NTAPI || {};
+                performanceMetricsArr.push(metrics);
+            } else {
+                // do this when Segment's analytics.js file is loaded into the application:
+                jQuery(document).trigger('onSegmentUpload', [metrics]); // upload each event as it occurs to Segment.com
```
Change to `onMetricsUpload`

Commit:
alter trigger event to be onMetricsUpload

This allows us to be more agnostic from an open source point of view.

Reviewed-By: Ryan Poplin <ryan.poplin@turner.com>
Fix: feature/CNNWE-828

crawls-turner commented on src/navigation-timing.js:
```diff
@@ -97,14 +97,28 @@ window.NTAPI = window.NTAPI || {};
-        } else if (window.debug) {
-            console.log('Navigation Timing Settings Definition Error. Metrics Type:', metrics.type, 'PostUrl:', window.NTAPI.PostUrl);
+        // NOTE: analytics -> {} & performanceMetricsArr -> [] are both initialized global's that should be setup via your client application; their purpose is to interface with Segment.com's analytics.js file that's loaded in on the client app
+        if (analytics && performanceMetricsArr) {
```
^ I just thought about this... what do you think about namespacing this "performanceMetricsArr" variable? maybe window.NTAPI.metricsArray ?

(... and also change in a PR for hope as well)

Commit:
add namespacing for this file

Now we use the 'window.NTAPI.metricsArray' naming convention.

Reviewed-By: Ryan Poplin <ryan.poplin@turner.com>
Fix: feature/CNNWE-828

Commit:
update the comment for the new global var

Alter the comment to not the new variable namespacing.

Reviewed-By: Ryan Poplin <ryan.poplin@turner.com>
Fix: feature/CNNWE-828
